101-hdinsight-rserver-ssh-password/azuredeploy.json

```diff
@@ -0,0 +1,180 @@
+			"metadata": {
+				"description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+			}
+		},
+		"clusterStorageAccountName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the storage account to be created and be used as the cluster's storage."
+			}
+		},
+		"clusterWorkerNodeCount": {
+			"type": "int",
+			"defaultValue": 4,
+			"metadata": {
+				"description": "The number of nodes in the HDInsight cluster."
+			}
+		}
+	},
+	"variables": {
+		"defaultApiVersion": "2015-05-01-preview",
+		"clusterApiVersion": "2015-03-01-preview"
+	},
+	"resources": [{
+		"name": "[concat(parameters('clusterName'),'/R-Server')]",
+		"type": "Microsoft.HDInsight/clusters/applications",
+		"apiVersion": "[variables('clusterApiVersion')]",
+		"dependsOn": ["[concat('Microsoft.HDInsight/clusters/',parameters('clusterName'))]"],
+		"properties": {
+			"marketPlaceIdentifier": "Microsoft.RServerForHDInsight.8.0.3",
+			"computeProfile": {
+				"roles": [{
+					"name": "edgenode",
+					"targetInstanceCount": 1,
+					"hardwareProfile": {
+						"vmSize": "Standard_D4_v2"
+					}
+				}]
+			},
+			"installScriptActions": [],
+			"uninstallScriptActions": [],
+			"httpsEndpoints": [],
+			"applicationType": "RServer"
+		}
+	},
+	{
+		"name": "[parameters('clusterStorageAccountName')]",
+		"type": "Microsoft.Storage/storageAccounts",
+		"location": "[parameters('location')]",
+		"apiVersion": "[variables('defaultApiVersion')]",
+		"dependsOn": [],
+		"tags": {
+			
+		},
+		"properties": {
+			"accountType": "Standard_LRS"
+		}
+	},
+	{
+		"name": "[parameters('clusterName')]",
+		"type": "Microsoft.HDInsight/clusters",
+		"location": "[parameters('location')]",
+		"apiVersion": "[variables('clusterApiVersion')]",
+		"dependsOn": ["[concat('Microsoft.Storage/storageAccounts/',parameters('clusterStorageAccountName'))]"],
+		"tags": {
+			
+		},
+		"properties": {
+			"clusterVersion": "3.4",
+			"osType": "Linux",
+			"tier": "Premium",
+			"clusterDefinition": {
+				"kind": "[parameters('clusterType')]",
+				"configurations": {
+					"gateway": {
+						"restAuthCredential.isEnabled": true,
+						"restAuthCredential.username": "[parameters('clusterLoginUserName')]",
+						"restAuthCredential.password": "[parameters('clusterLoginPassword')]"
+					}
+				}
+			},
+			"storageProfile": {
+				"storageaccounts": [{
+					"name": "[concat(parameters('clusterStorageAccountName'),'.blob.core.windows.net')]",
+					"isDefault": true,
+					"container": "[parameters('clusterName')]",
+					"key": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', parameters('clusterStorageAccountName')), variables('defaultApiVersion')).key1]"
+				}]
+			},
+			"computeProfile": {
+				"roles": [{
+					"name": "headnode",
+					"targetInstanceCount": "2",
+					"hardwareProfile": {
+						"vmSize": "Standard_D3_v2"
```

Comment:
Not for this version -- there will be a future improvement to HDInsight templates to have enums for node sizes.